### ISSUE TITLE
StanfordTokenizer failure

### ISSUE THREAD

**Issue by fievelk** [nltk/nltk#1584]
We are experiencing [`StanfordTokenizer` failures on our CI][1]:
```bash
Failed example:
    StanfordTokenizer().tokenize(s)
Exception raised:
    Traceback (most recent call last):
      File "/scratch/jenkins/python/python-3.4.2-x86_64/lib/python3.4/doctest.py", line 1324, in __run
        compileflags, 1), test.globs)
      File "<doctest nltk.tokenize.stanford.StanfordTokenizer[2]>", line 1, in <module>
        StanfordTokenizer().tokenize(s)
      File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/tokenize/stanford.py", line 68, in tokenize
        return self._parse_tokenized_output(self._execute(cmd, s))
      File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/tokenize/stanford.py", line 94, in _execute
        stdout=PIPE, stderr=PIPE)
      File "/scratch/jenkins/workspace/pull_request_tests/TOXENV/py34-jenkins/jdk/jdk8latestOnlineInstall/nltk/internals.py", line 134, in java
        raise OSError('Java command failed : ' + str(cmd))
    OSError: Java command failed : ['/opt/jdk/jdk8.latest/bin/java', '-mx1000m', '-cp', '/home/jenkins/third/stanford-postagger/stanford-postagger-3.7.0-sources.jar:/home/jenkins/third/stanford-postagger/stanford-postagger-3.7.0.jar:/home/jenkins/third/stanford-postagger/stanford-postagger-3.7.0-javadoc.jar:/home/jenkins/third/stanford-postagger/stanford-postagger.jar:/home/jenkins/third/stanford-postagger/lib/slf4j-simple.jar:/home/jenkins/third/stanford-postagger/lib/slf4j-api.jar', 'edu.stanford.nlp.process.PTBTokenizer', '-charset', 'utf8', '/tmp/tmptx_7hdjm']
```

@alvations I think that you have had some experience with this kind of problems previously. Could you please verify? I have the feeling that this could be related to some changes in the Stanford jar, but I could be wrong.

[1]:https://nltk.ci.cloudbees.com/job/pull_request_tests/lastCompletedBuild/TOXENV=py34-jenkins,jdk=jdk8latestOnlineInstall/testReport/nltk.tokenize/stanford/StanfordTokenizer/

**Comment by fievelk** [comment]
I think I might have found the problem, and it could be related to the `slf4j.jar` file currently shipped by Stanford CoreNLP. I opened an issue on their repo: stanfordnlp/CoreNLP#334.

I solved the problem (on my machine) substituting their `lib/slf4j-api.jar` with the [`1.7.12` version][1].

[1]:https://mvnrepository.com/artifact/org.slf4j/slf4j-api/1.7.12

**Comment by alvations** [comment]
@fievelk Sorry for the delayed response. 

An easy way out is to not specify any specific jar files and use the `nltk.internals.find_jars_within_path`. Given the main directory of Stanford corenlp, you can do something like https://github.com/nltk/nltk/blob/develop/nltk/tag/stanford.py#L60 and then append the stanford jars to the command line at https://github.com/nltk/nltk/blob/develop/nltk/tag/stanford.py#L93

But I think the cleaner way is to use the API from @dimazest on #1249

**Comment by fievelk** [comment]
Fixed by #1585
